src/main/java/pl/agawesolowska/ticketbookingapp/facade/BookingFacade.java

```diff
@@ -0,0 +1,81 @@
+package pl.agawesolowska.ticketbookingapp.facade;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import pl.agawesolowska.ticketbookingapp.model.dto.BookingRequestDTO;
+import pl.agawesolowska.ticketbookingapp.model.dto.BookingResultDTO;
+import pl.agawesolowska.ticketbookingapp.model.entity.Booking;
+import pl.agawesolowska.ticketbookingapp.model.entity.Customer;
+import pl.agawesolowska.ticketbookingapp.model.entity.Screening;
+import pl.agawesolowska.ticketbookingapp.model.entity.Seat;
+import pl.agawesolowska.ticketbookingapp.service.BookingService;
+import pl.agawesolowska.ticketbookingapp.service.CustomerService;
+import pl.agawesolowska.ticketbookingapp.service.ScreeningService;
+import pl.agawesolowska.ticketbookingapp.service.SeatService;
+import pl.agawesolowska.ticketbookingapp.util.DateTimeUtils;
+import pl.agawesolowska.ticketbookingapp.util.SeatReservationUtils;
+
+/**
+ * @author Aga Wesołowska
+ *
+ */
+@Component
+public class BookingFacade {
+
+	private final BookingService bookingService;
+	private final CustomerService customerService;
+	private final ScreeningService screeningService;
+	private final SeatService seatService;
+
+	@Autowired
+	public BookingFacade(BookingService bookingService, CustomerService customerService,
+			ScreeningService screeningService, SeatService seatService) {
+		this.bookingService = bookingService;
+		this.customerService = customerService;
+		this.screeningService = screeningService;
+		this.seatService = seatService;
+	}
+
+	public BookingResultDTO addCustomerBooking(BookingRequestDTO bookingRequestDTO) {
+
+		Screening screening = screeningService.getScreeningById(bookingRequestDTO.getScreening().getId());
+		screeningService.checkWhetherReservationIsPossible(screening);
+
+		SeatReservationUtils.checkWhetherSeatsAreNextToEachOther(bookingRequestDTO.getSeatReservations());
+
+		Set<Seat> seatsToBooking = new HashSet<>();
+		for (Seat seatReservation : bookingRequestDTO.getSeatReservations()) {
```

Comment:
jak to zrobić w stylu javy >=8 ?

Reply:
Możemy wykorzystać metodę forEach:
```
Set<Seat> seatsToBooking = new HashSet<>();
Set<Seat> seatReservations = bookingRequestDTO.getSeatReservations();
seatReservations.forEach(seatReservation -> {
	Seat seat = seatService.getSeatById(seatReservation.getId());
	seatService.checkWhetherSeatIsAvailable(seat);
	seatService.setPriceForTicket(seat, seatReservation.getTicketType());
	seatsToBooking.add(seat);
});
```
Ale niestety nie wiem jak napisać to lepiej, żeby zmiejszyć objętość kodu w bloku {}.